duckduckgo/Android
# Fireproof websites UI

Task/Issue URL: https://app.asana.com/0/488551667048375/1170688450957744/f
Tech Design URL: https://app.asana.com/0/481882893211075/1170688450957746/f
CC: 

## **Description**:
This PR implements all the UI work related to the feature fireproof websites. To make it functional and testable we are also including inside this PR the database creation to store fireproof websites.
List of included features:
- fireproof website option menu + confirmation snackbar
- fireproof websites settings option
- fireproof websites screen (list of fireproof websites)
- Room database to store fireproof websites
- improvements on bookmark screen (to keep consitency with fireproof websites)

## **Steps to test this PR**:
### Test 1
1. Fresh install
1. Visit any website
1. Open options menu and click on Fireproof website
1. Ensure snackbar appears with expected domain
1. Ensure Fireproof website in overflow menu is now disabled
1. Open Settings
1. Open Fireproof websites
1. Ensure fireproof website is listed and feature description appears below
1. Delete Fireproof website clicking on more options menu
1. Ensure list gets updated and only feature description appears
1. Go back to browser
1. Ensure Fireproof website in overflow menu is now enabled

### Test 2
1. Visit any non-fireproof website that starts with www.
1. Open options menu and click on Fireproof website
1. Ensure snackbar appears showing fireproof website without www. prefix
1. Open Settings
1. Open Fireproof websites
1. Ensure fireproof website is listed without www. prefix

### Test 3
1. Visit any non-fireproof website
1. Open options menu and click on Fireproof website
1. Ensure snackbar appears with expected domain
1. Click on "undo"
1. Ensure Fireproof website in overflow menu is still disabled

### Test 4
1. Visit any non-fireproof website
1. Open options menu and click on Fireproof website
1. Ensure snackbar appears with expected domain
1. Ensure Fireproof website in overflow menu is now disabled
1. Open a new tab
1. Visit a fireproof website
1. Ensure Fireproof website in overflow menu is disabled

### Test 5
1. Open a new tab
1. Ensure Fireproof website in overflow menu is disabled

### Test 6
1. Visit any non-fireproof website
1. Open options menu and click on Fireproof website
1. Ensure snackbar appears with expected domain
1. Ensure Fireproof website in overflow menu is now disabled
1. Navigate to another website
1. Navigate again to fireproof website
1. Ensure Fireproof website in overflow menu is enabled

### Test 7
1. checkout develop
1. Fresh install
1. Change to this branch
1. Install the app
1. Ensure database migration correctly executed
1. Perform any of the previous test to check functionality works

Additional test:
- Change between light/dark theme (only overflow menus are not rep


### **Screenshots:**
<img src="https://user-images.githubusercontent.com/4747865/80459896-5d4ebc80-8933-11ea-84f0-3d68319284c9.png" width="360px"/>
<img src="https://user-images.githubusercontent.com/4747865/80459938-68a1e800-8933-11ea-9916-e08f35f38796.png" width="360px"/>
<img src="https://user-images.githubusercontent.com/4747865/80459952-6c356f00-8933-11ea-83ae-326983c1a0f0.png" width="360px"/>
<img src="https://user-images.githubusercontent.com/4747865/80459967-72c3e680-8933-11ea-93f0-6fc74aa61a9f.png" width="360px"/>
<img src="https://user-images.githubusercontent.com/4747865/80569735-d31d5b80-89f9-11ea-8b63-602acf2ecc31.png" width="360px"/>
<img src="https://user-images.githubusercontent.com/4747865/80460006-80796c00-8933-11ea-8bd1-0012c0b97ee1.png" width="360px"/>

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
